src/UnifiedFFI/FFIFunctionParser.class.st

```diff
@@ -289,8 +289,8 @@ FFIFunctionParser >> returnType: aType [
 
 { #category : #accessing }
 FFIFunctionParser >> setOn: aStreamSource [
-
-	| functionDeclarationArray |
+	<haltOrBreakpointForDevelopment>
+	| functionDeclarationArray | 
 	functionDeclarationArray := aStreamSource isString ifTrue: [ | node |
 			node := RBParser parseExpression: '#(', aStreamSource, ')'.
 			node isLiteralNode ifFalse: [ self halt ].
```

Comment:
This looks like a debugging halt left behind, not like a halt set there on purpose.
Instead of adding \<haltOrBreakpointForDevelopment\> to paperwall this issue, I would replace the halt with an error like:
```Smalltalk
node isLiteralNode ifFalse: [ self error: 'A literal node is expected' ].
```

The PR looks good to me otherwise.

Reply:
I agree. Why not an assertion instead?

However, why not keeping the <haltOrBreakpointForDevelopment> in the dev versions?
That means that it should not be there in production images.

Reply:
Oups, this seems a leftover from a refactoring I did.
I'd prefer a non assertion exception. Either something like what @dupriezt proposes or maybe another kind of exception like `FFIUnsupportedFunctionDeclarationError` ?

Reply:
Good idea for the assertion.

> However, why not keeping the in the dev versions?
We have separate dev versions and production versions for pharo?

Reply:
Well yes.
Or, at least, there are the release and the dev.
Ok for the FFIUnsuporterd error.
For the pragma, either nobody ever leaves a halt in the code and it gets rejected every time, or we need the pragma so that it does not mess with the tools.